shell/browser/api/atom_api_app.cc

```diff
@@ -868,18 +868,21 @@ void App::SetAppLogsPath(mate::Arguments* args) {
 base::FilePath App::GetPath(mate::Arguments* args, const std::string& name) {
   bool succeed = false;
   base::FilePath path;
+
   int key = GetPathConstant(name);
-  if (key >= 0)
+  if (key >= 0) {
     succeed = base::PathService::Get(key, &path);
-  if (!succeed) {
-    if (name == "logs") {
-      args->ThrowError("Failed to get '" + name +
-                       "' path: setAppLogsPath() must be called first.");
-    } else {
-      args->ThrowError("Failed to get '" + name + "' path");
+    // If users try to get the logs path before setting a logs path,
+    // set the path to a sensible default and then try to get it again
+    if (name == "logs" && !succeed) {
+      SetAppLogsPath(args);
```

Comment:
This is probably safe, but it's a little weird to pass `args` here: (1) we actually _won't_ get our expected behavior if it happens to have anything in it, and (2) SetAppLogsPath() calls GetNext() and so the args->next_ field gets incremented.

It would be slightly safer to pass the address of a local temporary Args